qiime_studio/api/v1.py

```diff
@@ -115,3 +124,85 @@ def api_input_artifacts(plugin_name, workflow_name, input_name):
         if Artifact.load(artifact['path']).type <= input_type:
             input_artifacts.append(artifact)
     return jsonify({'input_artifacts': input_artifacts})
+
+
+@v1.route('/job/<plugin_name>/<workflow_name>', methods=['POST'])
+def execute_workflow(plugin_name, workflow_name):
+    plugin = PLUGIN_MANAGER.plugins[plugin_name]
+    workflow = plugin.workflows[workflow_name]
+
+    request_body = request.get_json()['jobData']
+
+    process_values = {}
+    for key, value in request_body.items():
+        if '-' in key:
+            type_, name = key.split('-')
+            process_values[name] = value
+
+    inputs = {
+        name: process_values[name]
+        for name in workflow.signature.inputs
+    }
+    parameters = {
+        name: process_values[name]
+        for name in workflow.signature.parameters
+    }
+    outputs = {
+        name: process_values[name]
+        for name in workflow.signature.outputs
+    }
+
+    def toggle_completion(process):
+        process_id = id(process)
+        completed_process = process.result()
+        if completed_process.returncode != 0:
+            __JOBS[process_id]['error'] = True
+            __JOBS[process_id]['message'] = \
+                completed_process.stderr.decode('utf-8')
+        __JOBS[process_id]['completed'] = True
+
+    now = '{:%Y-%b-%d %H:%M:%S}'.format(datetime.datetime.now())
+    process = SUBPROCESS_EXECUTOR(workflow, inputs, parameters, outputs)
+    process_id = id(process)
```

Comment:
I would shy away from calling this a `process_id` as it isn't actually the OS's `pid`, maybe call `process` `future_result`?